app/backend/src/couchers/servicers/auth.py

```diff
@@ -424,19 +424,31 @@ def CompletePasswordReset(self, request, context):
             else:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.INVALID_TOKEN)
 
-    # Only called if user doesn't have password
-    def ConfirmChangeEmailWithOldAddress(self, request, context):
+    def ConfirmChangeEmail(self, request, context):
         with session_scope() as session:
-            user = (
-                session.query(User)
-                .filter(User.old_email_token == request.change_email_token)
+            query = session.query(User).filter(User.id == context.user_id)
```

Comment:
The user should be able to verify the token without logging in (e.g. did something on computer and then clicks on email on the phone).

In general in open servicers (like auth is), we don't have `context.user_id` defined.

Reply:
Ya, that was one of the bugs which failed